rosbridge_server/src/rosbridge_server/websocket_handler.py

```diff
@@ -174,14 +173,16 @@ def on_close(self):
         self.incoming_queue.finish()
 
     def send_message(self, message, compression="none"):
+        cls = self.__class__
+
         if isinstance(message, bson.BSON):
             binary = True
         elif compression in ["cbor", "cbor-raw"]:
             binary = True
         else:
             binary = False
 
-        _io_loop.add_callback(partial(self.prewrite_message, message, binary))
+        asyncio.run_coroutine_threadsafe(self.prewrite_message(message, binary), cls.event_loop)
```

Comment:
If `event_loop` is `None` (its default value), will this error? Should we check?

Reply:
It probably will, the same with `node_handle`. I'm not happy with passing arguments through class variables but I just tried to adhere to how it is currently done.